app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.reg.email
+  before_save :email_down
+  validates :name, presence: true, length: { maximum: Settings.reg.maxname }
+  validates :email, format: { with: VALID_EMAIL_REGEX },
+                             presence: true, uniqueness: true, length: { maximum: Settings.reg.maxemail }
+  validates :password, presence: true, length: { minimum: Settings.reg.minipass }
+  
+  has_secure_password
+
+  private
+  def email_down 
+    email.downcase!
+  end
```

Comment:
```suggestion
  
  validates :name, presence: true, length: { maximum: Settings.reg.maxname }
  validates :email, format: { with: VALID_EMAIL_REGEX },
                             presence: true, uniqueness: true, length: { maximum: Settings.reg.maxemail }
  validates :password, presence: true, length: { minimum: Settings.reg.minipass }
  
  before_save :email_down
  
  has_secure_password

  private
  
  def email_down 
    email.downcase!
  end
```

Reply:
downcase! và downcase khác nhau ở điểm gì hả em?

Reply:
downcase! thì mới lưu được chữ in thường trong database, còn downcase thì không ạ